README.md

```diff
@@ -1,16 +1,658 @@
+        "text": "What's up?"
+      }
+    ],
+    "QmXrfqKNUpRiyyi8r3hpSZyZuG3S2MY9rTw1p8iEC9FNh5": {
+      "http": { 
+        "status": 200,
+        "body": "hello world"
+      },
+      "ms": 476
+    }
+  },
+  "nnc": "xyz",
+  "ext": {
+    "notes": "very receipt. such wow.",
+    "tags": ["dag-house", "fission"]
+  },
+  "sig": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt_VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA"
+ }
+```
+
+# 5 Promise Pipelining
+
+> Machines grow faster and memories grow larger. But the speed of light is constant and New York is not getting any closer to Tokyo. As hardware continues to improve, the latency barrier between distant machines will increasingly dominate the performance of distributed computation. When distributed computational steps require unnecessary round trips, compositions of these steps can cause unnecessary cascading sequences of round trips
+>
+> — [Mark Miller](https://github.com/erights), [Robust Composition](http://www.erights.org/talks/thesis/markm-thesis.pdf)
+
+At UCAN creation time, the UCAN MAY not yet have all of the information required to construct the next request in a sequence. Waiting for each request to complete before proceeding to the next action has a performance impact due to the amount of latency. [Promise pipelining](http://erights.org/elib/distrib/pipeline.html) is a solution to this problem: by referencing a prior invocation, a pipelined invocation can direct the executor to use the output of earlier invocations into the input of a later one. This liberates the invoker from waiting for each step.
+
+The authority to execute later actions often cannot be fully attenuated in advance, since the executor controls the reported output of the prior step in a pipeline. When choosing to use pipelining, the invoker MUST delegate capabilities for any of the possible outputs. If tight control is required to limit authority, pipelining SHOULD NOT be used.
+
+## 5.1 Promises
+
+## 5.1.1 Fields
+
+| Field           | Type         | Description                                                                           | Required | Default  |
+|-----------------|--------------|---------------------------------------------------------------------------------------|----------|----------|
+| `ucan/promised` | `CID or "/"` | The Invocation being referenced                                                       | Yes      |          |
+| `label`         | `String`     | The action's label. If the actions were implicit (`"run": "*"`), the the CID is used  | Yes      |          |
+| `selector`      | `String`     | The [JSONPath](https://datatracker.ietf.org/doc/draft-ietf-jsonpath-base/) expression | No       | `"$..*"` |
+
+The above table MUST be serialized as a tuple. In JSON, this SHOULD be represented as an array containing the values (but not keys) sequenced in the order they appear in the table.
+
+## 5.1.2 IPLD Schema
+
+``` ipldsch
+type Promise struct {
+  promised    Target
```

Comment:
Ok at this point I'm thinking the IPLD Schemas field names aren't intended to match the JSON field names :smile:

Reply:
Ah, indeed that's an overwritable option! I need to add that here. Thanks!

Reply:
Oh, actually because this gets serialized as a tuple, the names don't matter. I should remove the namespace field name from the table above.